interview-questions.md

```diff
@@ -15,54 +15,54 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
 
 1. What is git? What is the difference between git and Github?
 
-  Your answer:
+  Your answer: git is a file sharing system which allows for code sharing between different computers, either local to local or local to host. Github is a remote hosting space for code that is pushed over git, whereas git is just the code enabling the communication of code between local machine and remote site.
 
-  Researched answer:
+  Researched answer: git is a version control system in wide use amongst programmers. It was created by Linus Torvalds to track changes in source code. Whereas git is a version control system, Github is a cloud-based hosting service used which was designed as a git repository for primarily programming use.
 
 
 
 2. Which JavaScript operators will return a Boolean value?
 
-  Your answer:
+  Your answer: The Javascript operators =, ==, ===, <, >, etc will return Boolean values. Booleans are binary values of true and false that can be used when you need to return a value based on whether a certain condition is met or not.
 
-  Researched answer:
+  Researched answer: Javascript operators that will return a Boolean value are equality operators (==, ===), relational operators (>, <, >=, <=), and logical operators (and(&&), or(||), not(//)).
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: An index is a value corresponding to a logical ordered space within a string or array. Indexes start at 0 and proceed from there, so the index of the letter "e" in the string "Hello" would be 1. A value is the actual number, string, boolean, etc which occupies a space at a certain index. In our previous example, "e" is the value and 1 is the index.
 
-  Researched answer:
+  Researched answer: Not sure how I can paraphrase this one any further, I pretty much nailed it above ^.
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: Iteration is the repeating of a particular piece of code through a string, array, etc until a certain condition has been met. Iteration is usually accomplished with a for loop.
 
-  Researched answer:
+  Researched answer: The only thing I'll add here is that you can iterate until a certain condition is met OR a certain number of times.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: One person is the driver and the other is the navigator. The driver is actively coding and taking direction and input from the navigator. The navigator is not actively coding and is directing the code inputs of the driver as well as directing the troubleshooting of any errors.
 
-  Researched answer:
+  Researched answer: Crushed this one too, see above ^.
```

Comment:
Great job on this section!!  Your answers are very well thought out and thorough, excellent work!